grammars/jq.cson

```diff
@@ -81,11 +84,49 @@
       }
     ]
   'keyword':
-    'match': '(?<!\\.)\\b(and|as|break|catch|elif|else|end|foreach|if|import|include|label|module|or|reduce|then|try)(?!\\s*:)\\b'
+    'match': '(?<!\\.)\\b(and|as|break|catch|elif|else|empty|end|foreach|if|import|include|label|module|or|reduce|then|try)(?!\\s*:)\\b'
     'name': 'keyword.control.jq'
   'number':
     'match': '([0-9.]{2,}|[0-9]+)([eE][+-]?[0-9]+)?'
     'name': 'constant.numeric.jq'
+  'array':
+    'begin': '\\['
+    'captures':
+      '0':
+        'name': 'punctuation.bracket.square.jq'
+    'end': '\\]'
+    'name': 'meta.array.jq'
+    'patterns': [
+      {
+        'include': '#comment'
+      }
+      {
+        'include': '#string'
+      }
+      {
+        'begin': '\\('
+        'captures':
+          '0':
+            'name': 'punctuation.bracket.round.jq'
+        'end': '\\)'
+        'patterns': [
+          {
+            'include': '#self_in_round_brackets'
+          }
+        ]
+      }
+      {
+        'include': '$self'
```

Comment:
**Tip:** For recursive inclusions like this, it's best to define a single rule (named `#main`) and `include` that, rather than relying on the `$self` and `$base` variables.

**Reason:** Implementations have different ideas about what `$self` means when injections are involved, often differing in odd, subtle ways. Hence why I always write my grammars to have a single defined entry-point.